sql/src/main/java/org/apache/druid/sql/DirectStatement.java

```diff
@@ -301,6 +302,7 @@ public void cancel()
   public void close()
   {
     if (state != State.START && state != State.CLOSED) {
+      // super.close calls closeQuietly, which removes us from the sqlLifecycleManager.
```

Comment:
Nit: should we rely on `close`/`closeQuietly` to do the right thing?
Since we are anyway doing an explicit remove in `closeWithError`, I suppose we could do the same here too. Then we can just not override `closeQuietly`.

Alternatively, 
I think `closeWithQuietly` could be abstract and be more of a cleanup method which is called irrespective of whether closing with error or not.

Reply:
Ah, now that I read this again, I see that `closeQuietly` is called by both `close` and `closeWithError`. So we can do the second thing you suggested: only clean up the statement there. I just pushed up another commit with this change.